CppSource/Source/Services/Achievements/achievements.cpp

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pch.h"
+#include "xsapi/achievements_c.h"
+#include "achievements_helper.h"
+#include "achievements_taskargs.h"
+#include "achievements_state.h"
+#include "xbox_live_context_impl.h"
+
+using namespace xbox::services;
+using namespace xbox::services::achievements;
+
+xsapi_singleton* get_singleton_for_achievements()
+{
+    auto singleton = get_xsapi_singleton();
+
+    if (singleton->m_achievementsState == nullptr)
+    {
+        singleton->m_achievementsState = std::make_unique<XSAPI_ACHIEVEMENTS_STATE>();
+    }
```

Comment:
you could just do this once in the xsapi_singleton ctor and then get rid of this function and use the existing get_xsapi_singleton